src/Store.js

```diff
@@ -2,42 +2,114 @@ import {
   observable,
 } from 'mobx';
 
-class EmptyTransporter {}
+import DefaultItem from './Item';
+import EmptyTransporter from './Transporters/EmptyTransporter';
+import { ROLE, SOURCE, STATE } from './constants';
 
 export default class Store {
 
-  static transporterOptions = {
-    http: {
-      url: 'http',
-    },
+  @observable items = [];
+  @observable loaded = false;
+  // @observable deletedItems = [];
+  constructor({
+    Item = DefaultItem, schema,
+    transporter = new EmptyTransporter(),
+    clientStorage = new EmptyTransporter(),
+    options = { autoSave: true },
+  }) {
+    if (!schema) {
+      throw new Error('undefined schema');
+    }
+    this.transporter = transporter;
+    this.transporter.setEnvironment({ store: this, role: ROLE.TRANSPORTER });
+    this.clientStorage = clientStorage;
+    this.transporter.setEnvironment({ store: this, role: ROLE.CLIENT_STORAGE });
+    this.schema = schema;
+    this._options = options;
+    this._Item = Item;
+
+    this._isLoaded = {};
+    this._isLoaded.promise = new Promise((resolve, reject) => {
+      this._isLoaded.resolve = resolve;
+      this._isLoaded.reject = reject;
+    });
+
+    this.clientStorage.initialFetch([])
+    .then((csData) => {
+      csData.items.forEach((item) => {
+        item._transporterState = this._castItemTransporterState(item._transporterState);
+      });
+      this._createItems(csData.items, SOURCE.CLIENT_STORAGE);
+      return this.transporter.initialFetch(csData.items);
+    })
+    .then((tData) => {
+      tData.items.forEach((item) => {
+        item._transporterState = this._castItemTransporterState(item._transporterState);
+      });
+      this._createItems(tData.items, SOURCE.TRANSPORTER);
+      this._removeItems(tData.toDelete, SOURCE.TRANSPORTER);
+    })
+    .then(() => this._finishLoading());
+  }
 
-    socket: {
-      url: 'http',
-    },
+  // ///////////////////
+  // PUBLIC METHODS   //
+  // ///////////////////
 
-    indexedDb: {
-      tableName: '123',
-    },
-  };
+  create() {}
+  fetchAndCreate() {} // same as findOneOrFetch
+  fetch() {} // maybe? fetches again from the given SOURCE, defaults to transporter
 
-  constructor({ Item, Transporter, Storage }) {
-    this.Item = Item;
+  find() {} // returns array of items
+  findOne() {} // returns item or undefined
+  findOneOrFetch() {} // returns promise which resolves in harmonized promise
 
-    if (Transporter) {
-      this.transporter = new Transporter(this.constructor.transporterOptions[Transporter.name]);
-    } else {
-      this.transporter = new EmptyTransporter();
+  isLoaded() {
+    return this.loaded;
+  }
+
+  onceLoaded() {
+    return this._isLoaded.promise;
+  }
+
+  remove() {}
+
+  // ///////////////////
+  // PRIVATE METHODS  //
+  // ///////////////////
+
+  _castItemTransporterState(rawState) {
+    for (const state in STATE) {
```

Comment:
So for now I think this is ok until we figure out how we want to use lodash (there is a more functional way lodash can be used).